### ISSUE TITLE
Allow minimum contrast ratio to go in both directions if needed

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#3720]
Context: https://github.com/microsoft/vscode/issues/145973#issuecomment-1083451042

The minimum contrast ratio should make this black text white/light gray:

![image](https://user-images.githubusercontent.com/2193314/160929177-e4c2b87f-68ba-4387-8f64-b8fa794c9eae.png)

It does not currently because we only move the fg color away from the bg color, we should instead invert the foreground and go in the other direction if necessary.

**Comment by Tyriar** [comment]
An even worse problem is white on white where the text becomes invisible.
